components/registry-facade/pkg/registry/layersource.go

```diff
@@ -245,7 +260,35 @@ func NewStaticSourceFromImage(ctx context.Context, resolver remotes.Resolver, re
 		res = append(res, l)
 	}
 
-	return res, nil
+	var envs []EnvModifier
+	parsedEnvs := parseEnvs(cfg.Config.Env)
+	for _, name := range parsedEnvs.keys {
+		value := parsedEnvs.values[name]
+		if strings.HasPrefix(name, "GITPOD_APPEND_ENV_") {
```

Comment:
nit: I'd probably name the prefixes `GITPOD_ENV_SET_`, `GITPOD_ENV_APPEND_` and `GITPOD_ENV_PREPEND` and put them in a const section:
```Go
const (
    envPrefixSet = "GITPOD_ENV_SET_"
    envPrefixAppend = "GITPOD_ENV_APPEND_"
    envPrefixPrepend = "GITPOD_ENV_PREPEND_"
)
```